chat_shell/chat_shell/tools/sandbox/_base.py

```
@@ -0,0 +1,497 @@
+            bot_config: Bot configuration list
+            default_shell_type: Default shell type (ClaudeCode, Agno, etc.)
+            timeout: Execution timeout in seconds
+        """
+
+        # Injected dependencies - set when creating the tool instance
+        task_id: int = 0
+        subtask_id: int = 0
+        ws_emitter: Any = None
+        user_id: int = 0
+        user_name: str = ""
+        bot_config: list = []  # Bot config list [{shell_type, agent_config}, ...]
```

Ruff (0.14.14)

[warning] Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)